### ISSUE TITLE
ImageDataGenerator with no augmentation performs much worse compared to plain arrays

### ISSUE THREAD

**Issue by martinuray** [keras-team/keras-preprocessing#150]
Hi to all of you,

I am working on an image regression task, in a supervised manner. So I wanted to use the `ImageDataGenerator` to generate both input and target image. Here I tried several approaches. Mine is derived from the version shown in the [Keras Docs](https://keras.io/preprocessing/image/) (Image/Mask).

In order to verify my implementation, first I used the `ImageDataGenerator` without a parameter. My assumption here is that no major derivation, in comparison to the _baseline_ (without `ImageDataGenerator`), should be seen.
Unfortunately, this is not the case. The validation loss is somehow stable around a certain value and only slightly decreasing (see the image below).

![Comparison](https://i.stack.imgur.com/d0vhj.png)

The code I use for this task is the following. This is already altered by a lot of possible solutions by other users on [StackOverflow](https://stackoverflow.com/q/54302953/4153034).
```
np.random.shuffle(image_train)
split_seperator = int(image_train.shape[0]*validation_split)
image_val = image_train[0:split_seperator]
image_train = image_train[split_seperator:]

target_val = target_train[0:split_seperator]
target_train = target_train[split_seperator:]

data_gen_args = dict()
seed = 1

aug = ImageDataGenerator(**data_gen_args)
time_train_start = time.time()

# train the network
model.fit_generator(
    generator=aug.flow(image_train, target_train,
                       batch_size=settings.batch_size_training),
    validation_data=(image_val, target_val),
    steps_per_epoch=image_train.shape[0] // settings.batch_size_training,
    verbose=1, epochs=num_epochs,  callbacks=cb)
```

The code for the _baseline_ uses solely the stacks of images:
```
_ = model.fit(
    image_train, target_train,  # Train the model using the training set...
    batch_size=settings.batch_size_training,
    epochs=num_epochs, verbose=1,
    validation_split=validation_split, callbacks=cb)  # holding out fraction validation_split of the data for validation
```
Here I use `Keras-Preprocessing==1.0.5` .

As already mentioned, I posted this problem to [StackOverflow](https://stackoverflow.com/q/54302953/4153034), but unfortunately, this did not lead to a satisfying improvement by the given answers. 

Do you have any clue why I can not achieve the same results compared to my baseline?

Thanks in advance

**Comment by Dref360** [comment]
You do call `np.random.shuffle(image_train)`, but I do not see the same for operation for the target?

**Comment by amine1992** [comment]
I got the same problem. How did you solve it?

**Comment by martinuray** [comment]
See the message from @Dref360. It was all a stupid mistake.
